### PR TITLE
update eslint config & add JSDoc

### PR DESCRIPTION
**Pull request by gucong3000**
- add [JSDoc](https://www.npmjs.com/package/jsdoc)
  - add ·JSDoc· comments for [index.js](index.js), [lib/catFile.js](lib/catFile.js), [lib/diff.js](lib/diff.js)
- update eslint config
  - switch on [valid-jsdoc](http://eslint.org/docs/rules/valid-jsdoc)
  - remove rules that already defined in `eslint:recommended` , [no-undef](http://eslint.org/docs/rules/no-undef), [no-unused-vars](http://eslint.org/docs/rules/no-unused-vars)
- remove unused variables

### CODE REVIEW

**Comment by gucong3000** [comment]
I'd like to add support for `git.lsFile` & `git.blame` in next PR

File: test/commit.js [hunk 1]
```diff
@@ -58,7 +54,11 @@ module.exports = function(git, util) {
   it('should commit a file to the repo when appending paths is disabled', function(done) {
     var fakeFile = util.testOptionsFiles[4];
     exec('git add ' + fakeFile.path, {cwd: './test/repo/'},
-      function (error, stdout, stderr) {
+      function (error) {
+        if (error) {
+          done(error);
```
Comment (stevelacy) [inline test/commit.js]:
```js
if (error) return done(error);
```

File: test/commit.js [hunk 2]
```diff
@@ -78,7 +78,11 @@ module.exports = function(git, util) {
   it('should commit a file to the repo when passing multiple messages', function(done) {
     var fakeFile = util.testOptionsFiles[5];
     exec('git add ' + fakeFile.path, {cwd: './test/repo/'},
-      function (error, stdout, stderr) {
+      function (error) {
+        if (error) {
+          done(error);
+          return;
```
Comment (stevelacy) [inline test/commit.js]: Same as above, returning undefined and a callback will lead to issues.

**Comment by stevelacy** [comment]
I would have preferred you open an issue before adding JSDoc as my personal preference is to not use it, as right now there are only three files with JSDocs added. Making all the others not consistent.

What are the outputs of ls file and git blame? I do not think git blame is needed or should be added to this plugin.

**Comment by stevelacy** [comment]
Thank you, merged

**Pull request merged**